azuredevops/internal/service/memberentitlementmanagement/resource_group_entitlement.go

```diff
@@ -0,0 +1,378 @@
+	if d.Id() == "" {
+		return nil
+	}
+
+	groupEntitlementID := d.Id()
+	id, err := uuid.Parse(groupEntitlementID)
+	if err != nil {
+		return fmt.Errorf("Error parsing GroupEntitlement ID. GroupEntitlementID: %s. %v", groupEntitlementID, err)
+	}
+
+	clients := m.(*client.AggregatedClient)
+
+	_, err = clients.MemberEntitleManagementClient.DeleteGroupEntitlement(m.(*client.AggregatedClient).Ctx, memberentitlementmanagement.DeleteGroupEntitlementArgs{
+		GroupId: &id,
+	})
+
+	if err != nil {
+		return fmt.Errorf("Deleting group entitlement: %v", err)
+	}
+
+	// Also delete the org wise group if the group is Azure DevOps local, meaning
+	// most likely the local group was created by this resource
+	origin := d.Get("origin")
+	if origin == "vsts" {
+		err = clients.GraphClient.DeleteGroup(clients.Ctx, graph.DeleteGroupArgs{
+			GroupDescriptor: converter.String(d.Get("descriptor").(string)),
+		})
+
+		if err != nil {
+			return fmt.Errorf("Deleting Azure DevOps local group: %v", err)
+		}
+	}
+
+	return nil
+}
+
+func resourceGroupEntitlementUpdate(d *schema.ResourceData, m interface{}) error {
+	groupEntitlementID := d.Id()
+	id, err := uuid.Parse(groupEntitlementID)
+	if err != nil {
+		return fmt.Errorf("Parsing GroupEntitlement ID. GroupEntitlementID: %s. %v", groupEntitlementID, err)
+	}
+
+	accountLicenseType, err := converter.AccountLicenseType(d.Get("account_license_type").(string))
+	if err != nil {
+		return err
+	}
+	licensingSource, ok := d.GetOk("licensing_source")
+	if !ok {
+		return fmt.Errorf("Reading account licensing source for GroupEntitlementID: %s", groupEntitlementID)
+	}
+
+	clients := m.(*client.AggregatedClient)
+
+	patchResponse, err := clients.MemberEntitleManagementClient.UpdateGroupEntitlement(clients.Ctx,
+		memberentitlementmanagement.UpdateGroupEntitlementArgs{
+			GroupId: &id,
+			Document: &[]webapi.JsonPatchOperation{
+				{
+					Op:   &webapi.OperationValues.Replace,
+					From: nil,
+					Path: converter.String("/accessLevel"),
+					Value: struct {
+						AccountLicenseType string `json:"accountLicenseType"`
+						LicensingSource    string `json:"licensingSource"`
+					}{
+						string(*accountLicenseType),
+						licensingSource.(string),
+					},
+				},
+			},
+		})
+
+	if err != nil {
+		return fmt.Errorf("Updating group entitlement: %v", err)
+	}
+
+	result := *patchResponse.Results
+
+	if !*result[0].IsSuccess {
+		return fmt.Errorf("Updating group entitlement: %s", getGroupEntitlementAPIErrorMessage(&result))
+	}
+	return resourceGroupEntitlementRead(d, m)
+}
+
+var uuidRegexp = regexp.MustCompile("^[0-9a-fA-F]{8}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{12}$")
+
+func importGroupEntitlement(d *schema.ResourceData, m interface{}) ([]*schema.ResourceData, error) {
+	_, err := uuid.Parse(d.Id())
+	if err != nil {
+		upn := d.Id()
+		if !uuidRegexp.MatchString(upn) {
+			return nil, fmt.Errorf("Only UUID values can used for import [%s]", upn)
+		}
```

Comment:
```suggestion
		
```

Reply:
This is still required, uuid.parse doesn't valid the same as this regex. See the function `TestGroupEntitlement_Import_TestInvalidValue` in [resource_group_entitlement_test.go](https://github.com/stanleyz/terraform-provider-azuredevops/blob/114735f2953d6161a016f95b7e723a77d4ad26fb/azuredevops/internal/service/memberentitlementmanagement/resource_group_entitlement_test.go#L289)

Reply:
The `d.id()`  is from service which we can trust is a valid UUID. If the ID is not a valid UUID, there is a something wrong with the service

Reply:
normally yes, however this function is for importing existing resource, you can't guarantee the users will always have a valid UUID supplied, it's not a big deal but additional check with nice error message always helps.

Reply:
UUID already check by `_, err := uuid.Parse(d.Id())`, no need to verify twice

Reply:
cool, updated. That was my impression in the first look, double checked, there is a logic error in the test, not the `uuid.Parse`